### ISSUE TITLE
Unable to Insert BigDecimal 0 into NUMERIC Type Using JDBC/Hibernate

### ISSUE THREAD

**Issue by proddata** [crate/crate#16797]
### CrateDB version

5.9.0

### CrateDB setup information

Tarball
JDBC / Hibernate -> PG Protocol

### Problem description

There seems to be an issue while inserting data into a NUMERIC field Hibernate. Specifically, the BigDecimal value of 0 is causing a NumberFormatException. The error occurs during the process of persisting data to a CrateDB table, as shown in the logs and stack trace provided below:

```
Caused by: org.postgresql.util.PSQLException: ERROR: NumberFormatException in java.base/java.math.BigDecimal.<init>(BigDecimal.java:724)
Where: io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
```

<details><summary>Full error</summary>
<p>

```
] [insert into MyEntity (key2Value,myArray,"value",timestamp,myUUId) values (?,?,?,?,?)]
                at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:63)
                at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
                at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
                at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:134)
                at org.hibernate.engine.jdbc.mutation.internal.MutationExecutorSingleNonBatched.performNonBatchedOperations(MutationExecutorSingleNonBatched.java:55)
                at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.execute(AbstractMutationExecutor.java:55)
                at org.hibernate.persister.entity.mutation.InsertCoordinatorStandard.doStaticInserts(InsertCoordinatorStandard.java:194)
                at org.hibernate.persister.entity.mutation.InsertCoordinatorStandard.coordinateInsert(InsertCoordinatorStandard.java:132)
                at org.hibernate.persister.entity.mutation.InsertCoordinatorStandard.insert(InsertCoordinatorStandard.java:104)
                at org.hibernate.action.internal.EntityInsertAction.execute(EntityInsertAction.java:110)
                at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:644)
                at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:511)
                at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:414)
                at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)
                at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
                at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1429)
                at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:491)
                at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2354)
                at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1978)
                at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)
                at org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl.beforeCompletion(JtaTransactionCoordinatorImpl.java:336)
                at org.hibernate.resource.transaction.backend.jta.internal.synchronization.SynchronizationCallbackCoordinatorNonTrackingImpl.beforeCompletion(SynchronizationCallbackCoordinatorNonTrackingImpl.java:47)
                at org.hibernate.resource.transaction.backend.jta.internal.synchronization.RegisteredSynchronization.beforeCompletion(RegisteredSynchronization.java:37)
                at com.arjuna.ats.internal.jta.resources.arjunacore.SynchronizationImple.beforeCompletion(SynchronizationImple.java:52)
                at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.beforeCompletion(TwoPhaseCoordinator.java:348)
                at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.end(TwoPhaseCoordinator.java:66)
                at com.arjuna.ats.arjuna.AtomicAction.commit(AtomicAction.java:135)
                at com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.commitAndDisassociate(TransactionImple.java:1290)
                at com.arjuna.ats.internal.jta.transaction.arjunacore.BaseTransaction.commit(BaseTransaction.java:104)
                at io.quarkus.narayana.jta.runtime.NotifyingTransactionManager.commit(NotifyingTransactionManager.java:70)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.endTransaction(TransactionalInterceptorBase.java:406)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:175)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:107)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:38)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:61)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:32)
                at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(Unknown Source)
                at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
                at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:30)
                at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:27)
                at io.quarkus.arc.impl.InterceptedStaticMethods.aroundInvoke(InterceptedStaticMethods.java:22)
                at 

<... obfuscated ... >
org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
                at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
                at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
                at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
                at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
                at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
                at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
                at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
                at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
                at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
                at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
                at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
                at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
                at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
                at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
                at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
                at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
                at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
                at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
                at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
                at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
                at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
                at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
                at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:198)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:169)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:93)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:58)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:141)
                at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:57)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
                at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:85)
                at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:124)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:99)
                at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:94)
                at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:63)
                at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
                at java.base/java.lang.reflect.Method.invoke(Method.java:580)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
                at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
                at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
                at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
                at jdk.proxy1/jdk.proxy1.$Proxy4.stop(Unknown Source)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:200)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:132)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
                at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
                at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:121)
                at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
                at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: org.postgresql.util.PSQLException: ERROR: NumberFormatException in java.base/java.math.BigDecimal.<init>(BigDecimal.java:724)
  Where: io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
```

</p>
</details> 

```
[insert into MyEntity (key2Value,myArray,"value",timestamp,myUUId) values (?,?,?,?,?)]
                at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:63)
                at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
                at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
                at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:134)
```

---

Test provided by @BaurzhanSakhariev in PostgresITest

```java
@Test
public void test_numeric_type() throws Exception {
    properties.setProperty(PGProperty.BINARY_TRANSFER_ENABLE.getName(), "1700");
    try (Connection conn = DriverManager.getConnection(url(RW), properties)) {
        conn.createStatement().executeUpdate(
            "CREATE TABLE t (" +
                "   l NUMERIC(18,2)" +
                ") " +
                "WITH (number_of_replicas = 0)");

        PreparedStatement preparedStatement = conn.prepareStatement(
            "INSERT INTO t (l) VALUES (?)");
        preparedStatement.setBigDecimal(1, BigDecimal.ZERO);
        preparedStatement.executeUpdate();
    } catch (BatchUpdateException e) {
        throw e.getNextException();
    }
}

```

```
org.postgresql.util.PSQLException: ERROR: NumberFormatException in java.base/java.math.BigDecimal.<init>(BigDecimal.java:724)
  Where: io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
```

### Steps to Reproduce

```sql
CREATE TABLE IF NOT EXISTS "doc"."myentity" (
   "myuuid" VARCHAR(36) NOT NULL,
   "key2value" OBJECT(DYNAMIC),
   "myarray" ARRAY(VARCHAR(255)),
   "value" NUMERIC(38, 2),
   "timestamp" TIMESTAMP WITH TIME ZONE,
   PRIMARY KEY ("myuuid")
)
```

```
{
  "key2Value": {
    "key1": "value1",
    "key2": "value2"
  },
  "timestamp": 1729005941.611091412,
  "myValue": 0,
  "myArray": ["item1", "item2", "item3"]
}
```

### Actual Result

```
Caused by: org.postgresql.util.PSQLException: ERROR: NumberFormatException in java.base/java.math.BigDecimal.<init>(BigDecimal.java:724)
Where: io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
```

### Expected Result

Insert works

**Comment by matriv** [comment]
Thx @proddata for spotting and reporting this! also @BaurzhanSakhariev for the reproduction itest!
It has been fixed, and the fix will be available with the upcoming hotfix releases for 5.9 and 5.8.

**Comment by mackerl** [comment]
Thanks for the awesome quick fix and turnaround. Greetings from SSI Schäfer.
BR Mario
